File: src/main/webapp/WEB-INF/src/js/app/administration/students/StudentService.js
module.exports = function ($http) {
    var _this = this;

    _this.findPage = function (groupUuid) {
        var params = {
            size: 10000000,
            page: 0,
            groupUuid: groupUuid
        };
        return $http.get('/api/students', {params: params}).then(function (response) {
            return response.data;
        });
    };

    _this.findOne = function (uuid) {
        var params = {};
        return $http.get('/api/students/' + uuid, {params: params}).then(function (response) {
            return response.data;
        });
    };

    _this.create = function (student) {
        return $http.post('/api/students', student).then(function (response) {
            return response.data;
        });
    };

    _this.update = function (student) {
        return $http.put('/api/students/' + student.uuid, student).then(function (response) {
            return response.data;
        })
    };

    _this.delete = function (student) {
        return $http.delete('/api/students/' + student.uuid);
    }
};

File: src/main/java/uz/com/journal/app/student/StudentService.java
package uz.com.journal.app.student;


import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import uz.com.journal.core.NonDeletableService;
import uz.com.platform.app.users.User;

public interface StudentService extends NonDeletableService<Student> {
    Page<Student> findPage(StudentFetch fetch, StudentCriteria criteria, Pageable pageable, User currentUser);

    Student findOne(String uuid, StudentFetch fetch, User currentUser);

    Student create(StudentItem item, User currentUser);

    Student update(String uuid, StudentItem item, User currentUser);
}
